src/Infrastructure/IO/src/ESMCI_IO_NetCDF_Utils.C

```diff
@@ -0,0 +1,138 @@
+// $Id$
+//
+// Earth System Modeling Framework
+// Copyright (c) 2002-2025, University Corporation for Atmospheric Research,
+// Massachusetts Institute of Technology, Geophysical Fluid Dynamics
+// Laboratory, University of Michigan, National Centers for Environmental
+// Prediction, Los Alamos National Laboratory, Argonne National Laboratory,
+// NASA Goddard Space Flight Center.
+// Licensed under the University of Illinois-NCSA License.
+//
+//==============================================================================
+#define ESMC_FILENAME "ESMCI_IO_NetCDF_Utils.C"
+//==============================================================================
+
+// single blank line to make protex happy.
+// BOP
+
+// EOP
+//-------------------------------------------------------------------------
+//
+//  !DESCRIPTION:
+//
+//  The code in this file implements utility functions for NetCDF.
+//
+//-------------------------------------------------------------------------
+
+// associated header file
+#include "ESMCI_IO_NetCDF_Utils.h"
+
+using namespace std;
+
+//-----------------------------------------------------------------------------
+// leave the following line as-is; it will insert the cvs ident string
+// into the object file for tracking purposes.
+static const char *const version = "$Id$";
+//-----------------------------------------------------------------------------
+
+//-------------------------------------------------------------------------
+#undef ESMC_METHOD
+#define ESMC_METHOD "ESMCI::NetCDFUtils::ncerrToEsmcRc"
+int ESMCI::NetCDFUtils::ncerrToEsmcRc(int ncerror)
+{
+   if (ncerror == NC_NOERR)
+      return ESMF_SUCCESS;
+   else
+      return ESMF_FAILURE;
+} // end ncerrToEsmcRc
+
+//-------------------------------------------------------------------------
+#undef ESMC_METHOD
+#define ESMC_METHOD "ESMCI::NetCDFUtils::ncToEsmcType"
+ESMC_TypeKind_Flag ESMCI::NetCDFUtils::ncToEsmcType(nc_type ncTypeVal)
+{
+
+   ESMC_TypeKind_Flag esmcTypeVal = ESMF_NOKIND;
+
+#if defined(ESMF_NETCDF) || defined(ESMF_PNETCDF)
```

Comment:
I considered a few ways to do the conditional compilation here and in esmcToNcType. Based on a quick code analysis, I think we need these functions to be compiled even if we're not using netcdf or pnetcdf, but the return value is unimportant in those cases. I don't love putting essentially the whole body of the function inside an `#if`, but this seemed like the best way to do it - though I'm open to other suggestions.

(As a side note: I'd love if we could just say that netcdf is required, or have some intermediate layer between netcdf and esmf that we use from esmf to avoid a direct dependency, so we could avoid having `#if`s like this in the code, because I find they make it much harder to read, write and modify code.)